RIOT-OS/RIOT
# boards/nrf52: introduce shared dependencies in common/nrf52 and update boards

### Contribution description

When reviewing #11769, I found that softdevice package and nrf802154 dependencies could be moved in common files under `boards/common/nrf52` using the `Makefile.nrf52<family>.dep` pattern. This is a bit similar to the `Makefile.nrf52832.features` added in #11769.

So this PR is proposing this change and applies it to nrf52840 based boards as well as some nrf52832 based boards. Note that for some nrf52832 boards, the softdevice is now added (nrf52832-mdk, acd52832).

Maybe the naming the new files could be improved (use `softdevice` and `nrf802154` instead of `nrf52832` and `nrf52840` instead ? This would also allow to add an nrfmin dep file for example).
Suggestions are welcome.




### Testing procedure

Build and flash `gnrc_networking` on related boards and verify the expected dependencies are pulled in (softdevice on nrf52832, nrf802154 on nrf52840)




### Issues/PRs references

None
